### ISSUE TITLE
metricset sample value reported as NaN

### ISSUE THREAD

**Issue by kruskall** [elastic/apm-agent-java#3281]
## Describe the bug

As part of the ops KPI review we've been investigating the following errors in ecs logs:

```
decode error: data read error: v2.metricsetRoot.Metricset: v2.metricset.Samples: v2.metricsetSampleValue.Value: readNumberAsString: invalid number, error found in #10 byte of ...|:{"value":"NaN"},"to|..., bigger context ...|"jsp"},"samples":{"tomcat.servlet.error":{"value":"NaN"},"tomcat.servlet.request.count":{"value":0},|...
```

`NaN` is not a valid metric sample value. The spec expects a number.

This was observed with the following agents:

- elasticapm-java/1.19.0
- elasticapm-java/1.18.0

## Steps to reproduce

Unfortunately, I don't have steps to reproduce 

## Expected behavior

Data is reported correctly and ingestion succeed

## Debug logs

This is part of the ops KPI review so we only have the error logs in elastic cloud

**Comment by SylvainJuge** [comment]
As discussed internally, here the best option would be to update the agents as the `1.18.x` or `1.19.x` are very old.

**Comment by simitt** [comment]
Agree that updating the agents should be the path forward, if this is already fixed in newer version.

**Comment by SylvainJuge** [comment]
I can't pinpoint which exact version this should have been fixed in, but I am quite confident it will solve the issue here:
- the version 1.19.0 was released in November 2020, so more than 2.5 years ago.
- there was some bugfixes around the same period, including AFTER the 1.19.0 version: https://github.com/elastic/apm-agent-java/issues?q=is%3Aclosed+NaN
- this issue hasn't been seen in the field for more recent versions.

**Comment by SylvainJuge** [comment]
This issue has been fixed in recent agent versions, and there isn't much we can do besides asking users to upgrade, so I think we can close this now.
